src/main/java/io/confluent/copycat/hdfs/HdfsSinkTask.java

```diff
@@ -43,37 +43,51 @@ public HdfsSinkTask() {
   public void start(Properties props) {
     try {
       HdfsSinkConnectorConfig connectorConfig = new HdfsSinkConnectorConfig(props);
-      hdfsWriter = new HdfsWriter(connectorConfig);
-      context.resetOffset(hdfsWriter.getPreviousOffsets());
+      assignment = context.assignment();
+      hdfsWriter = new HdfsWriter(connectorConfig, assignment);
+      recover(assignment);
+      resetOffsets();
     } catch (ConfigException e) {
-      throw new CopycatRuntimeException(
+      throw new CopycatException(
```

Comment:
Should we be doing more to clean up here? I'm thinking that if `start()` encounters an error, then we should probably be giving up any resources we've acquired so far. In particular, if we got the leases for some of the WALs, shouldn't we give them up since the task won't be in a state to make use of them unless `start()` was reinvoked?